src/java.base/share/classes/java/util/regex/MatchResult.java

```diff
@@ -74,7 +77,44 @@ public interface MatchResult {
      *          If there is no capturing group in the pattern
      *          with the given index
      */
-    public int start(int group);
+    int start(int group);
+
+    /**
+     * Returns the start index of the subsequence captured by the given
+     * <a href="Pattern.html#groupname">named-capturing group</a> during the
+     * previous match operation.
+     *
+     * @param  name
+     *         The name of a named-capturing group in this matcher's pattern
+     *
+     * @return  The index of the first character captured by the group,
+     *          or {@code -1} if the match was successful but the group
+     *          itself did not match anything
+     *
+     * @throws  IllegalStateException
+     *          If no match has yet been attempted,
+     *          or if the previous match operation failed
+     *
+     * @throws  IllegalArgumentException
+     *          If there is no capturing group in the pattern
+     *          with the given name
+     *
+     * @implNote
+     * The default implementation of this method makes use of the map returned
+     * by {@link #namedGroups()}. It is thus sufficient to override
+     * {@link #namedGroups()} for this method to work. However, overriding this
+     * method directly might be preferable for performance or for other reasons.
```

Comment:
This `@implNote` text, is repeated in three different methods. Consider moving this to the class specification. It might make it a bit easier for implementors to see a central overview instead of having this information in each method.